File: restore_index.js
const fs = require('fs');
const content = fs.readFileSync('C:\\\\Users\\\\bilal\\\\SARACAPP\\\\SARACAPPV3\\\\backs\\\\1107_0330\\\\app1_source.md', 'utf8');
const lines = content.split('\\n');
let inTarget = false;
let out = [];
for (let i = 0; i < lines.length; i++) {
  if (!inTarget && (lines[i].includes('## File: app1\\src\\main\\index.ts') || lines[i].includes('## File: app1/src/main/index.ts'))) {
    inTarget = true;
    i++; // skip ```typescript
    continue;
  }
  if (inTarget && lines[i].trim() === '```') {
    break;
  }
  if (inTarget) {
    out.push(lines[i]);
  }
}
fs.writeFileSync('C:\\\\Users\\\\bilal\\\\SARACAPP\\\\SARACAPPV3\\\\app1\\\\src\\\\main\\\\index.ts', out.join('\\n'));
console.log('Restored index.ts with length: ' + out.length);

File: restore_index.py
import re

with open('C:\\\\Users\\\\bilal\\\\SARACAPP\\\\SARACAPPV3\\\\backs\\\\1107_0330\\\\app1_source.md', 'r', encoding='utf-8') as f:
    content = f.read()

# find index.ts
pattern = r'## File: app1\\src\\main\\index\.ts\n```typescript\n(.*?)```'
match = re.search(pattern, content, re.DOTALL)
if match:
    with open('C:\\\\Users\\\\bilal\\\\SARACAPP\\\\SARACAPPV3\\\\app1\\\\src\\\\main\\\\index.ts', 'w', encoding='utf-8') as f:
        f.write(match.group(1))
    print(f"Extracted {len(match.group(1))} bytes.")
else:
    print("Not found")
